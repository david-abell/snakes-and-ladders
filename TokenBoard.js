"use-strict";

class TokenBoard {
  canvas;

  context;

  constructor(containerEl, size) {
    if (!containerEl) throw new Error(`Missing container el`);
    if (!size) throw new Error(`Missing size parameter`);

    this.boardContainer = containerEl;
    this.boardSize = size;
    this.init();
  }

  setCanvas() {
    if (!document.getElementById("token-board")) {
      const canvas = document.createElement("canvas");
      canvas.setAttribute("id", "token-board");
      this.boardContainer.appendChild(canvas);
    }
    this.canvas = document.getElementById("token-board");
    this.canvas.width = this.boardSize;
    this.canvas.height = this.boardSize;
    this.context = this.canvas.getContext("2d");
  }

  draw(x, y, radius, fillStyle) {
    const circle = new Path2D();
    circle.arc(x, y, radius, 0, 2 * Math.PI);
    this.context.fillStyle = fillStyle;
    this.context.beginPath();
    this.context.fill(circle);
    return circle;
  }

  clear() {
    this.context.clearRect(0, 0, this.canvas.width, this.canvas.height);
  }

  init() {
    this.setCanvas();
  }
}

export default TokenBoard;
