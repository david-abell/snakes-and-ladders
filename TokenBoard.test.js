import TokenBoard from "./TokenBoard.js";

describe("drawGrid", () => {
  test("", () => {});
});

describe("init", () => {
  beforeEach(() => {
    document.body.innerHTML = `<div id="board-container"></div>`;
  });

  test("throw an error if size parameter missing", () => {
    const containerEl = document.getElementById("board-container");
    const boardConstructor = () => {
      // eslint-disable-next-line no-new
      new TokenBoard(containerEl);
    };
    expect(boardConstructor).toThrowError(/Missing size parameter/);
  });

  test("throw an error if container parameter missing", () => {
    const boardConstructor = () => {
      // eslint-disable-next-line no-new
      new TokenBoard();
    };
    expect(boardConstructor).toThrowError(/Missing container el/);
  });

  test("should create canvas element in board-container", () => {
    const containerEl = document.getElementById("board-container");
    // eslint-disable-next-line no-unused-vars
    const game = new TokenBoard(containerEl, 800);
    const gameBoard = document.getElementById("token-board");
    expect(gameBoard).toBeTruthy();
  });
});

describe("draw", () => {
  let containerEl;
  let game;
  beforeEach(() => {
    containerEl = document.createElement("div");
    game = new TokenBoard(containerEl, 800);
  });

  test(`should return the player token`, () => {
    const playerToken = game.draw(15, 15, 15, "#FE7E6D");
    expect(playerToken instanceof Path2D).toBeTruthy();
  });

  test(`should draw a player token on the board context`, () => {
    // eslint-disable-next-line no-unused-vars
    const playerToken = game.draw(15, 10, 15, "#FE7E6D");
    // eslint-disable-next-line no-underscore-dangle
    const events = game.context.__getEvents();
    expect(events).toMatchSnapshot();
  });
});
