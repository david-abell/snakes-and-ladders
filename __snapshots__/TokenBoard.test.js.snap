// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`draw should draw a player token on the board context 1`] = `
Array [
  Object {
    "props": Object {
      "value": "#fe7e6d",
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "fillStyle",
  },
  Object {
    "props": Object {},
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "beginPath",
  },
  Object {
    "props": Object {
      "fillRule": "nonzero",
      "path": Array [
        Object {
          "props": Object {
            "anticlockwise": false,
            "endAngle": 6.283185307179586,
            "radius": 15,
            "startAngle": 0,
            "x": 15,
            "y": 15,
          },
          "transform": Array [
            1,
            0,
            0,
            1,
            0,
            0,
          ],
          "type": "arc",
        },
      ],
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "fill",
  },
  Object {
    "props": Object {
      "value": "#fe7e6d",
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "fillStyle",
  },
  Object {
    "props": Object {},
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "beginPath",
  },
  Object {
    "props": Object {
      "fillRule": "nonzero",
      "path": Array [
        Object {
          "props": Object {
            "anticlockwise": false,
            "endAngle": 6.283185307179586,
            "radius": 15,
            "startAngle": 0,
            "x": 15,
            "y": 10,
          },
          "transform": Array [
            1,
            0,
            0,
            1,
            0,
            0,
          ],
          "type": "arc",
        },
      ],
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "fill",
  },
]
`;
